runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/WorkItemStatusClient.java

```diff
@@ -116,6 +117,10 @@ public synchronized void setWorker(
   /** Return the {@link WorkItemServiceState} resulting from sending an error completion status. */
   public synchronized WorkItemServiceState reportError(Throwable e) throws IOException {
     checkState(!finalStateSent, "cannot reportUpdates after sending a final state");
+    if (wasAskedToAbort) {
+      LOG.info("Service already asked to abort work item, not reporting ignored progress.");
+      return null;
```

Comment:
Add `@Nullable` to the return type. Are you sure it is safe? I don't see other nulls returned. I guess checking is turned off for this class...

Reply:
Added.

This method has a `return execute(status)` and `execute` returns `@Nullable WorkItemServiceState`. There are also tests that return null, which I discovered with a previous variant of this PR that didn't handle that case. Any Java Object can be null.

Reply:
Thanks for adding it. This will be one less for the person who eventually enables checkerframework.